cpp/src/io/orc/reader_impl.cu

```diff
@@ -549,6 +551,68 @@ class aggregate_orc_metadata {
   }
 };
 
+void snappy_decompress(device_span<gpu_inflate_input_s> comp_in,
```

Comment:
feels like we could merge this with snappy_decompress in Parquet reader, but I'm fine with leaving this for another PR.

Reply:
I realized that but once cuIO's internal decompressors are removed completely, there's a lot more to refactor around this that I deferred it for then.